lms/models/lti_event.py

```diff
@@ -0,0 +1,94 @@
+from enum import Enum
+from functools import lru_cache
+
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy import insert
+
+from lms.db import BASE
+
+
+class LTIEventType(BASE):
+    __tablename__ = "lti_event_type"
+
+    class Type(str, Enum):
+        LAUNCH = "launch"
+        DEEP_LINKING = "deep_linking"
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    type = sa.Column(
+        "type",
+        sa.Enum(
+            Type,
+            # In order to maintain maximum flexibility we will only enforce the
+            # type on the Python side, and leave the Postgres side open as a plain
+            # VARCHAR
+            native_enum=False,
+            create_constraint=False,
+            validate_strings=True,
+            # Without a length SQLAlchemy will constrain it to the longest value
+            # we happen to have right now, which could limit us in future
+            length=64,
+            # Use the string values, not the keys to persist the values
+            values_callable=lambda obj: [item.value for item in obj],
```

Comment:
This is an annoying pile 'o' code where if you get one part wrong it behaves strangely. If we see this for a third time, I think it's definitely time for a convenience wrapper in `db.py` or something.